src/Sulu/Component/Webspace/Loader/BaseXmlFileLoader.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Webspace\Loader;
+
+use Symfony\Component\Config\Loader\FileLoader;
+use Symfony\Component\Config\Util\XmlUtils;
+
+/**
+ * This is the base class for webspace xml file loaders. It offers an supports method, which will not only check if the
+ * file is in the xml format, but also if the loader supports the given schema. The supported schema has to be
+ * overriden by the base classby defining its own SCHEMA_URI constant.
+ */
+abstract class BaseXmlFileLoader extends FileLoader
+{
+    const SCHEMA_IDENTIFIER = 'http://schemas.sulu.io/webspace/webspace';
+
+    const SCHEMA_URI = '';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($resource, $type = null)
+    {
+        if (!is_string($resource) || 'xml' !== pathinfo($resource, PATHINFO_EXTENSION)) {
+            return false;
+        }
+
+        $document = XmlUtils::loadFile($resource);
```

Comment:
Haven't found any fast better solution to do this. The file has to be read to know which loader should be used, and caching the file would mean that this service introduces state, which should be avoided. Also, this operation is only called when writing the cache, so it shouldn't be a big deal anyway.